tslib.js

```diff
@@ -64,9 +64,14 @@ var __classPrivateFieldSet;
         function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
 
     __extends = function (d, b) {
+        if (b === null) {
+            d.prototype = Object.create(b);
+            return;
+        }
         extendStatics(d, b);

```

Comment:
Why not just do this instead?
```suggestion
        if (b !== null) extendStatics(d, b);
```

Reply:
Because it’s necessary to also add the `d.prototype.constructor` property in the `b === null` case.

Reply:
Your proposed change at the time I wrote that comment also did not set the `constructor` when `b === null`. Given your point about `constructor`, I'd probably recommend something simpler (and with fewer characters overall):
```ts
__extends = function (d, b) {
    if (typeof b !== "function" && b !== null)
        throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
    if (b) extendStatics(d, b);
    function __() {}
    (d.prototype = b ? (__.prototype = b.prototype, new __()) : Object.create(b)).constructor = d;
};
```

A few notes:
- I'm using `if (b)` rather than `if (b !== null)` since the null check happens on the first line and a function's "truthiness" can't be overridden through `valueOf`, `Symbol.toPrimitive`, or a `Proxy`
- I swapped the condition on the last line from `b === null` to `b`, and therefore swapped the order of expressions in the conditional.
- I moved the `constructor` assignment out of the `__` prototype helper and onto the last line.

Reply:
It might be better to fold the `extendStatics` call inside the `b ?` conditional to lower the amount of `ToBoolean(b)` checks necessary:
```js
function __extends(d, b) {
	if (typeof b !== "function" && b !== null)
		throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
	function __() {}
	(d.prototype = b
		? (extendStatics(d, b), __.prototype = b.prototype, new __())
		: Object.create(b)
	).constructor = d;
}
```